### PR TITLE
Kycfile contract

### PR DESCRIPTION
**Pull request by lawlawlaw**
Include contract in kycfile.

### CODE REVIEW

File: src/validation.cpp [hunk 1]
```diff
@@ -84,7 +84,8 @@ bool fHavePruned = false;
 bool fPruneMode = false;
 bool fIsBareMultisigStd = DEFAULT_PERMIT_BAREMULTISIG;
 bool fRequireStandard = true;
-bool fContractInTx = false;
+bool fContractInTx = DEFAULT_CONTRACT_INTX;
```
Comment (nkostoulas) [inline src/validation.cpp]: DEFAULT_CONTRACT_IN_TX

File: src/validation.cpp [hunk 2]
```diff
@@ -84,7 +84,8 @@ bool fHavePruned = false;
...
+bool fContractInKYCFile = DEFAULT_CONTRACT_INKYCFILE;
```
Comment (nkostoulas) [inline src/validation.cpp]: DEFAULT_CONTRACT_IN_KYCFILE

File: src/chainparams.cpp [hunk 3]
```diff
@@ -169,6 +169,7 @@ class CCustomParams : public CChainParams {
         fRequireStandard = GetBoolArg("-frequirestandard", false);
         fEmbedContract = GetBoolArg("-embedcontract", DEFAULT_EMBED_CONTRACT);
         fContractInTx = GetBoolArg("-contractintx", DEFAULT_CONTRACT_INTX);
+        fContractInKYCFile = GetBoolArg("-contractinkycfile", DEFAULT_CONTRACT_INKYCFILE);
```
Comment (nkostoulas) [inline src/chainparams.cpp]: Do we need another config param? Is it different from fContractInTx? If fContractInKYCFile is false and fContractInTx is true what happens? 

Should we add some test cases for this or are we always going to include the contract in the KYC file from now on?

Comment (lawlawlaw) [reply src/chainparams.cpp]: It's true by default so we'll leave it as an option for now.

File: src/validation.cpp (hunk 1 above)
Comment (lawlawlaw) [reply src/validation.cpp]: Ok

File: src/validation.cpp (hunk 2 above)
Comment (lawlawlaw) [reply src/validation.cpp]: Ok

**Comment by lawlawlaw** [comment]
It takes a short while for the whitelist to resync after restart.
I don't think all sleeps are required, but it would require several docker
build restarts to determine.

**Pull request merged**